### PR TITLE
feat(ACL-20) - implement payments auth flow

### PR DESCRIPTION
**Pull request by tl-mauro-franchi**
Added:
- new `StartAuhorizationFlow` request object to `CreatePayment` (see [doc](https://docs.truelayer.com/reference/create-payment#:~:text=AUTHORIZATION_FLOW-,OBJECT,-provider_selection))
- new `Authorizing` response object for `CreatePayment` (see [doc](https://docs.truelayer.com/reference/create-payment#:~:text=AUTHORIZATION%20REQUIRED-,AUTHORIZING,-object))

### CODE REVIEW

File: examples/MvcExample/Controllers/HomeController.cs
Comment (dili91) [inline examples/MvcExample/Controllers/HomeController.cs:1]: Can we safely ignore the deprecation warning on this file?

Comment (tl-mauro-franchi) [reply examples/MvcExample/Controllers/HomeController.cs:1]: yes as this controller is for the example app. 
As we discussed, we can revisit the examples project later.

File: test/TrueLayer.AcceptanceTests/PaymentTests.cs [hunk 1]
```diff
@@ -48,6 +52,39 @@ public async Task Can_create_payment(CreatePaymentRequest paymentRequest)
...
+                    providerSelection, new SchemeSelection.Preselected(),
+                    new Redirect(new Uri("http://localhost:3000/callback")))
+            );
+
+            var response = await _fixture.Client.Payments.CreatePayment(
+                paymentRequest, idempotencyKey: Guid.NewGuid().ToString());
+
+            response.StatusCode.ShouldBe(HttpStatusCode.Created);
+            response.Data.IsT3.ShouldBeTrue();
+            CreatePaymentResponse.Authorizing authorizing = response.Data.AsT3;
```
Comment (dili91) [inline test/TrueLayer.AcceptanceTests/PaymentTests.cs:76]: worth defining an `AsAuthorizing()` utility on the entity?

File: test/TrueLayer.AcceptanceTests/PaymentTests.cs [hunk 2]
```diff
@@ -48,6 +52,39 @@ public async Task Can_create_payment(CreatePaymentRequest paymentRequest)
...
+            authorizing.Id.ShouldNotBeNullOrWhiteSpace();
+            authorizing.ResourceToken.ShouldNotBeNullOrWhiteSpace();
+            authorizing.User.ShouldNotBeNull();
+            authorizing.User.Id.ShouldNotBeNullOrWhiteSpace();
+            authorizing.Status.ShouldBe("authorizing");
+
+            string hppUri = _fixture.Client.Payments.CreateHostedPaymentPageLink(
+                authorizing.Id, authorizing.ResourceToken, new Uri("https://redirect.mydomain.com"));
+            hppUri.ShouldNotBeNullOrWhiteSpace();
```
Comment (dili91) [inline test/TrueLayer.AcceptanceTests/PaymentTests.cs:85]: Maybe we should check that the authorization_flow is not null and includes a redirect next action?

File: test/TrueLayer.AcceptanceTests/PaymentTests.cs (hunk 1 above)
Comment (tl-mauro-franchi) [reply test/TrueLayer.AcceptanceTests/PaymentTests.cs:76]: surely we can add it but I would like to do it for every union in a follow-up Pr if you don't mind

Comment (dili91) [reply test/TrueLayer.AcceptanceTests/PaymentTests.cs:76]: Sure!

File: test/TrueLayer.AcceptanceTests/PaymentTests.cs (hunk 2 above)
Comment (tl-mauro-franchi) [reply test/TrueLayer.AcceptanceTests/PaymentTests.cs]: done

**Review by dili91 — APPROVED** [review]

**Pull request merged**